### ISSUE TITLE
[java] AvoidArrayLoops false positive on double assignment

### ISSUE THREAD

**Issue by Androbin** [pmd/pmd#1167]
**Affects PMD Version:** 5.8.0

**Rule:** [AvoidArrayLoops](https://pmd.github.io/latest/pmd_rules_java_performance.html#avoidarrayloops)

**Description:**
The rule sees an assignment to `foo[ i ]` and an add-assignment from `foo[ i ]`.
The rule thinks it sees an assignment from `foo[ i ]` to `foo[ i ]`.

**Code Sample demonstrating the issue:**

```java
public class Test {
  public static void main( String[] args ) {
    double[] foo = new double[ 100 ];
    double bar = 0.0;
    
    for ( int i = 0; i < exps.length; i++ ) {
      double value = Math.random();
      foo[ i ] = Math.exp( value );
      bar += foo[ i ];
    }
  }
}
```

**Running PMD through:** Eclipse Plugin 4.0.15

**Comment by Fernal73** [comment]
```
package pmdtests;

import java.util.Random;

public enum ArrayCopy {
  ;
  private static String[] words = {"the",
                            "hello",
                            "goodbye",
                            "packt",
                            "java",
                            "thread",
                            "pool",
                            "random",
                            "class",
                            "main"};

  public static void main(String[] args) {
    String[][] document = new String[10][100];
    Random random = new Random();

    for (int i = 0; i < 10; i++) {
      for (int j = 0; j < 100; j++) {
        int index = random.nextInt(words.length);
        document[i][j] = words[index];
      }
    }
  }
}
```

**Comment by Java8753** [comment]
another false positive similar to these, which solarlint correctly ignores:
(eclipse plugin: PMD Plug-in	4.39.0.v20220930-1439)

        for ( i = 1 ; i < 10 ; i++ )
        {
            temp [ i ] += temp [ i-1 ] ;
        }
